pkg/ingester/label_names_and_values.go

```diff
@@ -136,6 +136,25 @@ func labelValuesCardinality(
 		if err != nil {
 			return err
 		}
+
+		// Check if empty value for label is allowed according to matchers. This
+		// is an optimization only to avoid fetching postings, since the posting
+		// list would be empty anyway.
+		emptyValueAllowedByMatchers := true
+		for _, matcher := range matchers {
+			if matcher.Name != lblName {
+				continue
+			}
+			if !matcher.Matches("") {
+				emptyValueAllowedByMatchers = false
+				break
+			}
+		}
+
+		if emptyValueAllowedByMatchers {
+			lblValues = append(lblValues, "")
```

Comment:
this would hugely increase the scope of the endpoint when requesting negative matchers (`pod!="X"`). So the ingester will have to also find all series without the `pod` label, which means taking the union of all the lists for all values of `pod` and subtract them from the list of postings for this call. 

Is it an option to just subtract the sum of all label values from the number of all series that match the matchers and add this value to the result instead?